clang/lib/CIR/CodeGen/CIRGenExprConst.cpp

```diff
@@ -996,8 +997,9 @@ namespace {
 /// A struct which can be used to peephole certain kinds of finalization
 /// that normally happen during l-value emission.
 struct ConstantLValue {
+  // FIXME(cir): does it make sense to have both parent and child type here?
   using SymbolTy = mlir::SymbolRefAttr;
-  llvm::PointerUnion<mlir::Value, SymbolTy> Value;
+  llvm::PointerUnion<mlir::Value, SymbolTy, mlir::Attribute> Value;
```

Comment:
To your FIXME above: I don't think it makes sense, please remove SymbolTy in favor of `mlir::Attribute` and dyn_cast the attr to `mlir::SymbolRefAttr` whenever you need specific semantics!